safetrace-app/Services/Permissions/LocationProviding.swift

```diff
@@ -0,0 +1,114 @@
+import CoreLocation
+import Foundation
+import ReactiveSwift
+import SafeTrace
+import UIKit
+
+protocol ReactiveLocationProtocol {
+    var current: SignalProducer<CLLocation?, Never> { get }
+    var authorizationStatus: SignalProducer<CLAuthorizationStatus, Never> { get }
+}
+
+protocol LocationProviding {
+    var reactive: ReactiveLocationProtocol { get }
+    var currentLocation: CLLocation? { get }
+    var currentAuthorization: CLAuthorizationStatus { get }
+    func requestAlwaysLocationPermissions()
+}
+
+class ReactiveLocation: ReactiveLocationProtocol {
+    var current: SignalProducer<CLLocation?, Never>
+    var authorizationStatus: SignalProducer<CLAuthorizationStatus, Never>
+
+    fileprivate let _location = MutableProperty<CLLocation?>(nil)
+    fileprivate let _authorizationStatus = MutableProperty<CLAuthorizationStatus>(.notDetermined)
+
+    init() {
+        current = _location.producer
+        authorizationStatus = _authorizationStatus.producer
+    }
+}
+
+class LocationProvider: NSObject, LocationProviding {
+
+    fileprivate let _reactive = ReactiveLocation()
+    fileprivate let locationManager = CLLocationManager()
+    fileprivate let decimalPrecision = 5 // 5 is precise to around 1 meter at the equator. More precision just causes unnecessary app updates
+    private let desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+    private let distanceFilter: CLLocationDistance = 10
```

Comment:
@maxmamis I'm not familiar with the implications with these precision values, but I'm assuming that the contact tracer prefers accurate info. These can be adjusted if needed